src/state/predictions/index.ts

```diff
@@ -156,6 +165,82 @@ export const fetchHistory = createAsyncThunk<{ account: string; bets: Bet[] }, {
   },
 )
 
+export const fetchNodeHistory = createAsyncThunk<{ account: string; bets: Bet[] }, string>(
+  'predictions/fetchNodeHistory',
+  async (account) => {
+    const userRounds = await fetchLatestUserRounds(account)
+
+    if (!userRounds) {
+      return { account, bets: [] }
+    }
+
+    const epochs = Object.keys(userRounds).map((epochStr) => Number(epochStr))
+    const roundData = await getRoundsData(epochs)
+
+    // Turn the data from the node into an Bet object that comes from the graph
+    const bets: Bet[] = roundData.reduce((accum, round) => {
+      const reduxRound = serializePredictionsRoundsResponse(round)
+      const ledger = userRounds[reduxRound.epoch]
+      const ledgerAmount = ethers.BigNumber.from(ledger.amount)
+      const closePrice = round.closePrice ? parseFloat(formatUnits(round.closePrice, 8)) : null
+      const lockPrice = round.lockPrice ? parseFloat(formatUnits(round.lockPrice, 8)) : null
+
+      const getRoundPosition = () => {
+        if (!closePrice) {
+          return null
+        }
+
+        return round.closePrice.gt(round.lockPrice) ? BetPosition.BULL : BetPosition.BEAR
+      }
+
+      return [
+        ...accum,
+        {
+          id: null,
+          hash: null,
+          amount: parseFloat(formatUnits(ledgerAmount)),
+          position: ledger.position,
+          claimed: ledger.claimed,
+          claimedAt: null,
+          claimedHash: null,
+          claimedBNB: 0,
+          claimedNetBNB: 0,
+          createdAt: null,
+          updatedAt: null,
+          block: 0,
+          round: {
+            id: null,
+            epoch: round.epoch.toNumber(),
+            failed: false,
+            startBlock: null,
+            startAt: round.startTimestamp ? round.startTimestamp.toNumber() : null,
+            startHash: null,
+            lockAt: round.lockTimestamp ? round.lockTimestamp.toNumber() : null,
+            lockBlock: null,
+            lockPrice,
+            lockHash: null,
+            lockRoundId: round.lockOracleId ? round.lockOracleId.toString() : null,
+            closeRoundId: round.closeOracleId ? round.closeOracleId.toString() : null,
+            closeHash: null,
+            closeAt: null,
+            closePrice,
+            closeBlock: null,
+            totalBets: 0,
+            totalAmount: parseFloat(formatUnits(round.totalAmount)),
+            bullBets: 0,
+            bullAmount: parseFloat(formatUnits(round.bullAmount)),
+            bearBets: 0,
+            bearAmount: parseFloat(formatUnits(round.bearAmount)),
+            position: getRoundPosition(),
+          },
+        },
+      ]
```

Comment:
I made the decision to transform the node data into the shape of a `Bet` so that when data does come through from the subgraph we can simply merge it to update.